libdnf/sack/query.cpp

```diff
@@ -2010,30 +2026,29 @@ Query::filterExtras()
 
     // create query with available packages without non-modular excludes. As a extras should be
     // considered anso packages in non-active modules
-    Query query_available(pImpl->sack, HY_IGNORE_NON_MODULAR_EXCLUDES);
+    Query query_available(pImpl->sack, Query::ExcludeFlags::IGNORE_NON_MODULAR_EXCLUDES);
     query_available.addFilter(HY_PKG_REPONAME, HY_NEQ, HY_SYSTEM_REPO_NAME);
     query_available.apply();
 
     auto resultAvailable = query_available.pImpl->result.get();
     Id id_available = -1;
 
     // make vector of available packages names
-    std::vector<Id> names;
-    names.reserve(resultAvailable->size());
+    std::vector<Solvable *> namesArch;
+    namesArch.reserve(resultAvailable->size());
     while ((id_available = resultAvailable->next(id_available)) != -1) {
-        Solvable * s_available = pool_id2solvable(pool, id_available);
-        names.push_back(s_available->name);
+        namesArch.push_back(pool_id2solvable(pool, id_available));
     }
-    std::sort(names.begin(), names.end());
+    std::sort(namesArch.begin(), namesArch.end(), NameArchSolvableSorter);
     Id id_installed = -1;
     auto resultInstalled = query_installed.pImpl->result.get();
     
-    auto end = names.end();
+    auto end = namesArch.end();
```

Comment:
I don't think it's worth a variable for that one use on line 2051.

Reply:
It is used in loop, and I don't want to depend on optimizations during the build.

Reply:
Interesting point. I've never seen this optimization (and I've seen a lot of `.end()` calls), but I've found it discussed on stack overflow (https://github.com/rpm-software-management/ci-dnf-stack/pull/610 - note it's quite old). Standard mandates end() is of O(1) complexity. From what I've gathered from the sources, the end() call are a couple of inlined functions and a creation of an iterator object from a pointer.

Without some performance data this is just subjective opinion, but I'd say here it's a speculative optimization that won't have a noticeable impact on performance (in comparison with the rest of the loop), but does affect readability (somewhat). That is if it doesn't get optimized by the compiler.

I'll leave it up to you, but I'd prefer if we in general didn't make the code more complex by adding these things unless we know they in fact have an impact...

Reply:
Ok, it was removed